src/pages/StockConfiguration/logisticHelper.js

```diff
@@ -0,0 +1,224 @@
+            dataElements.push(element?.dataElement.id)
+        )
+
+        programData.push({
+            id: program.id,
+            teas,
+            dataElements,
+        })
+    })
+
+    programData.forEach((program) => {
+        const itemCodeList = []
+        const itemDescriptionList = []
+        const stockOnHandList = []
+        const distributedToList = []
+        const stockDistributedList = []
+        const stockCorrectedList = []
+        const stockDiscardedList = []
+
+        if (itemCode && itemDescription) {
+            program.teas.forEach((tea) => {
+                findSaveElement(tea, itemCode, itemCodeList)
+                findSaveElement(tea, itemDescription, itemDescriptionList)
+            })
+        }
+
+        if (
+            stockOnHand &&
+            distributedTo &&
+            stockDistributed &&
+            stockCorrected &&
+            stockDiscarded
+        ) {
+            program.dataElements.forEach((element) => {
+                findSaveElement(element, stockOnHand, stockOnHandList)
+                findSaveElement(element, distributedTo, distributedToList)
+                findSaveElement(element, stockDistributed, stockDistributedList)
+                findSaveElement(element, stockCorrected, stockCorrectedList)
+                findSaveElement(element, stockDiscarded, stockDiscardedList)
+            })
+        }
+
+        program.itemCode = itemCodeList
+        program.itemDescription = itemDescriptionList
+        program.stockOnHand = stockOnHandList
+        program.distributedTo = distributedToList
+        program.stockDistributed = stockDistributedList
+        program.stockCorrected = stockCorrectedList
+        program.stockDiscarded = stockDiscardedList
+    })
+
+    return {
+        programData,
+        itemCode,
+        stockOnHand,
+        itemDescription,
+        distributedTo,
+        stockDistributed,
+        stockCorrected,
+        stockDiscarded,
+    }
+}
+
+const findSaveElement = (element, list, saveList) => {
+    const foundElement = list.find((e) => e.id === element)
+    if (foundElement) {
+        saveList.push(foundElement)
+    }
+}
+const getDataElementType = (elements) => {
+    const stockOnHand = []
+    const distributedTo = []
+    const stockDistributed = []
+    const stockCorrected = []
+    const stockDiscarded = []
+
+    elements?.forEach((element) => {
+        const { valueType } = element
+
+        if (valueType === NUMBER) {
+            stockOnHand.push(element)
+        } else if (valueType === TEXT) {
+            distributedTo.push(element)
+        } else if (valueType === INTEGER_POSITIVE) {
+            stockDistributed.push(element)
+            stockDiscarded.push(element)
+        }
+
+        if (
+            valueType === INTEGER ||
+            valueType === INTEGER_POSITIVE ||
+            valueType === NUMBER
+        ) {
+            stockCorrected.push(element)
+        }
```

Comment:
IMHO the `if-else` pattern is hard to read. Maybe a switch or a lookup table pattern is more suitable in this case. WDYT?